### PR TITLE
Remove StreamForwarder from CLI tests

### PR DESCRIPTION
**Pull request by piotrpMSFT**
Fixes for CLI tests #4913, a big part of #4660

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
@dotnet-bot test this please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@livarcocc @krwq @jgoshi @jonsequitur can you guys take another look at this PR? There were some got'chas in the asynchronous operations that caused a lot of code churn.

File: test/dotnet-new.Tests/GivenThatIWantANewApp.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+using FluentAssertions;
+
+namespace Microsoft.DotNet.New.Tests
+{
+    public class GivenThatIWantANewApp : TestBase
+    {
+        [Fact]
+        public void When_dotnet_new_is_invoked_mupliple_times_it_should_fail()
```
Comment (livarcocc) [inline test/dotnet-new.Tests/GivenThatIWantANewApp.cs]: Remove snake casing.

**Comment by piotrpMSFT** [comment]
@MattGertz

**Pull request merged**